processor/src/main/java/org/mapstruct/ap/internal/util/workarounds/SpecificCompilerWorkarounds.java

```diff
@@ -115,4 +114,34 @@ public static TypeElement replaceTypeElementIfNecessary(Elements elementUtils, T
         }
         return element;
     }
+
+    /**
+     * Workaround for Bugs in the Eclipse implementation of {@link Types#asMemberOf(DeclaredType, Element)}.
+     *
+     * @see <a href="https://bugs.eclipse.org/bugs/show_bug.cgi?id=382590">Eclipse Bug 382590</a>
+     * @see <a href="https://bugs.eclipse.org/bugs/show_bug.cgi?id=481555">Eclipse Bug 481555</a>
+     */
+    static TypeMirror asMemberOf(Types typeUtils, ProcessingEnvironment env, DeclaredType containing, Element element) {
+        TypeMirror result = null;
+        Exception lastException = null;
+        try {
+            try {
+                result = typeUtils.asMemberOf( containing, element );
+            }
+            catch ( IllegalArgumentException e ) {
```

Comment:
So this assumes the IAE only ever occurs with Eclipse, right? Probably it's good enough, though I'd personally rather check whether it's actually Eclipse and otherwise pass on the IAE so to not obfuscate it.